### ISSUE TITLE
(Solved) Another Module Not Found Error

### ISSUE THREAD

**Issue by mlewus** [RomelTorres/alpha_vantage#165]
I am trying to use the library on a Raspberry Pi and I am getting the Module Not Found error on ”from alpha_vantage.timeseries import TimeSeries”.

I did an install of the release and also the development alpha_vantage package downloaded from github on a Pi 4 running Raspbian v10 (Buster) and kernel 4.19.75. My Python version is 3.7.3 and pip version is 19.3.1. The library installs and builds without warnings or errors - console output follows. I am using a python virtual environment.

All I’m trying to do is use the python shell interpreter to import the module. It fails with “Module Not Found.” Executing a dir(alpha_vantage) on the module indicates that it is not exporting “.timeseries”. running a small demo app which I have named alphav.py also fails.  I have looked at all of the other similar github issues but did not find a solution. But I am new to python so I am probably doing something stupid.  Can you help?

Path:
(env) pi@magic-mirror:~ $ echo $PATH
/home/pi/env/bin:/home/pi/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/games:/usr/games

Versions:
(env) pi@magic-mirror:~ $ lsb_release -a
No LSB modules are available.
Distributor ID: Raspbian
Description:    Raspbian GNU/Linux 10 (buster)
Release:        10
Codename:       buster
(env) pi@magic-mirror:~ $ uname -a
Linux magic-mirror 4.19.75-v7l+ #1270 SMP Tue Sep 24 18:51:41 BST 2019 armv7l GNU/Linux

Lib install:
(env) pi@magic-mirror:~ $ git clone https://github.com/RomelTorres/alpha_vantage.git
Cloning into 'alpha_vantage'...
remote: Enumerating objects: 1560, done.
remote: Total 1560 (delta 0), reused 0 (delta 0), pack-reused 1560
Receiving objects: 100% (1560/1560), 923.40 KiB | 1.68 MiB/s, done.
Resolving deltas: 100% (902/902), done.
(env) pi@magic-mirror:~ $ pip -V
pip 19.3.1 from /home/pi/env/lib/python3.7/site-packages/pip (python 3.7)
(env) pi@magic-mirror:~ $ pip install -e alpha_vantage
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Obtaining file:///home/pi/alpha_vantage
Requirement already satisfied: requests in ./env/lib/python3.7/site-packages (from alpha-vantage==2.1.2) (2.22.0)
Requirement already satisfied: chardet<3.1.0,>=3.0.2 in ./env/lib/python3.7/site-packages (from requests->alpha-vantage==2.1.2) (3.0.4)
Requirement already satisfied: idna<2.9,>=2.5 in ./env/lib/python3.7/site-packages (from requests->alpha-vantage==2.1.2) (2.8)
Requirement already satisfied: certifi>=2017.4.17 in ./env/lib/python3.7/site-packages (from requests->alpha-vantage==2.1.2) (2019.9.11)
Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in ./env/lib/python3.7/site-packages (from requests->alpha-vantage==2.1.2) (1.25.6)
Installing collected packages: alpha-vantage
  Running setup.py develop for alpha-vantage
Successfully installed alpha-vantage

Run in Python interactive shell:
(env) pi@magic-mirror:~ $ python
Python 3.7.3 (default, Apr  3 2019, 05:39:12)
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import alpha_vantage
>>> dir(alpha_vantage)
['__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__path__', '__spec__']
>>> from alpha_vantage.timeseries import TimeSeries
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'alpha_vantage.timeseries'
>>>

Demo app, contents of alphav.py:
(env) pi@magic-mirror:~ $ cat alphav.py
from alpha_vantage.timeseries import TimeSeries
key = (KEY REMOVED)
ts = TimeSeries(key)
amzn, meta = ts.get_daily(symbol='AMZN')
print(amzn['2019-11-10'])

Running app causes same failure:
(env) pi@magic-mirror:~ $ python alphav.py
Traceback (most recent call last):
  File "alphav.py", line 1, in <module>
    from alpha_vantage.timeseries import TimeSeries
ModuleNotFoundError: No module named 'alpha_vantage.timeseries'
(env) pi@magic-mirror:~ $

Thanks for any help you can offer.
Mark

**Comment by PatrickAlphaVantage** [comment]
Could you ask this question on StackOverflow or another forum? I'm curious about the answer as well, but it's not really an issue with the repo here.

**Comment by mlewus** [comment]
Yes, I can, and please excuse my ignorance, but why is it not an issue with the contents of the repo?

**Comment by PatrickAlphaVantage** [comment]
All good! Not ignorance, you're learning. Questions are great. And actually... now that you mention it, it could be an issue with the repo, but it's more likely something to do with your setup... I'm not too familiar with raspberry pi. 

I'm sure on StackOverflow you'll get a better answer. Do you mind posting the link here when you ask it?

**Comment by mlewus** [comment]
I did a little more work on it and there appears to be something wrong with “pip install -e alpha_advantage” (from the repo Readme.md). If I run my test from the alpha_vantage directory I can import timeseries and it works. Running it from anywhere else cause it to fail so it appears that it is not installing correctly. Pip thinks it is installed correctly but...

I tried the same process using a different Linux box (a Synology server) running python 3.5, with the same result. I am going to look a bit more into pip and if I can’t solve it myself I’ll post a question on stackoverflow. Thanks for your help.

**Comment by PatrickAlphaVantage** [comment]
Great, keep us posted?

Just another thought, it might have something to do with how your virtual environment is set up.

**Comment by mlewus** [comment]
You are probably right, as the library is there, but Python can't seem to find it. I'll let you know when I solve it, or failing that, when I post to Stackoverflow.

**Comment by mlewus** [comment]
Solved.

Pip was inadvertently installing the library into the wrong version of the Python interpreter. I found an existing discussion on this topic here: [https://stackoverflow.com/questions/32680081/importerror-after-successful-pip-installation/34797844](url).

I found another discussion about how permissions can cause similar issues here:
[https://stackoverflow.com/questions/14295680/unable-to-import-a-module-that-is-definitely-installed/37955665](url)

All working fine now. Thanks for pointing me in the right direction.
